examples/gatsby/gatsby-spa/src/utils/useContentfulClient.ts

```diff
@@ -0,0 +1,18 @@
+
+import { useMemo } from 'react';
+import { createClient } from 'contentful';
+
+export const useContentfulClient = () => {
+  const client = useMemo(() => {
+    const clientConfig = {
+      space: process.env.GATSBY_CTFL_SPACE || '',
+      environment: process.env.GATSBY_CTFL_ENVIRONMENT || '',
+      host: 'cdn.contentful.com',
+      accessToken: process.env.GATSBY_CTFL_ACCESS_TOKEN || '',
```

Comment:
probably be good to grab a `isPreview` param from the querystring and make the host dynamic based on if they are trying to pull up preview mode or delivery mode here.